docs/env_var_reference.rst

```diff
@@ -0,0 +1,76 @@
+Environment Variables
+=====================
+CumulusCI has environment variables that are useful when CumulusCI is being run inside of web applications such as MetaCI, MetaDeploy, and Metecho.
+The following is a reference list of available environment variables that can be set.
+
+
+
+``CUMULUSCI_AUTO_DETECT`` 
+
+    Set this environment variable to autodetect branch and commit information from ``HEROKU_TEST_RUN_BRANCH`` and ``HEROKU_TEST_RUN_COMMIT_VERSION`` environment variables.
```

Comment:
Might be worth adding "This is relevant when running CumulusCI inside Heroku CI, and is not commonly used."